src/Native/System.Native/pal_io.cpp

```diff
@@ -149,7 +149,8 @@ static void ConvertFileStatus(const struct stat_& src, FileStatus* dst)
 extern "C" int32_t Stat(const char* path, FileStatus* output)
 {
     struct stat_ result;
-    int ret = stat_(path, &result);
+    int ret;
+    while (CheckIo(ret = stat_(path, &result)));
```

Comment:
I can't find a reference to EINTR in man pages on Linux, OS X, or FreeBSD for stat.  I wonder if it's actually possible.  That said, it doesn't really hurt to have the extra checks.